modules/runner/src/main/java/org/apache/ignite/internal/configuration/DistributedConfigurationUpdater.java

```diff
@@ -31,37 +31,34 @@ public class DistributedConfigurationUpdater implements IgniteComponent {
 
     private static final IgniteLogger LOG = Loggers.forClass(DistributedConfigurationUpdater.class);
 
-    private final CompletableFuture<ConfigurationPresentation<String>> clusterCfgPresentation = new CompletableFuture<>();
+    private final ClusterManagementGroupManager cmgMgr;
 
-    public void setDistributedConfigurationPresentation(ConfigurationPresentation<String> presentation) {
-        clusterCfgPresentation.complete(presentation);
+    private final ConfigurationPresentation<String> presentation;
+
+    public DistributedConfigurationUpdater(ClusterManagementGroupManager cmgMgr, ConfigurationPresentation<String> presentation) {
+        this.cmgMgr = cmgMgr;
+        this.presentation = presentation;
     }
 
-    /**
-     * Applies changes to the cluster configuration when {@link DistributedConfigurationUpdater#clusterCfgPresentation}
-     * is complete.
-     *
-     * @param configurationToApply Cluster configuration that should be applied.
-     * @return Future that will be completed when cluster configuration is updated.
-     */
-    public CompletableFuture<Void> updateConfiguration(String configurationToApply) {
-        return clusterCfgPresentation.thenCompose(presentation -> presentation.update(configurationToApply))
+    @Override
+    public void start() {
+        cmgMgr.clusterConfigurationToUpdate()
+                .thenCompose(action -> {
+                    if (action.configuration() != null) {
+                        return presentation.update(action.configuration()).thenApply(ignored -> action);
+                    } else {
+                        return CompletableFuture.completedFuture(action);
+                    }
+                })
+                .thenCompose(action -> action.nextAction().get())
```

Comment:
Why do we need to run the `nextAction` even if no configuration was provided in the previous step?